src/UI/Implementation/Component/Signal.php

```diff
@@ -15,12 +16,50 @@ class Signal implements \ILIAS\UI\Component\Signal {
 	protected $id;
 
 	/**
+	 * @var array
+	 */
+	protected $options = array();
+
+	/**
 	 * @param string $id
 	 */
 	public function __construct($id) {
 		$this->id = $id;
 	}
 
+	/**
+	 * @inheritdoc
+	 */
+	public function getId() {
+		return $this->id;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getOptions() {
+		return $this->options;
+	}
+
+	/**
+	 * @param string $key
+	 * @param mixed $value
+	 */
+	protected function addOption($key, $value) {
+		$this->options[$key] = $value;
+	}
+
+	/**
+	 * @param string $key
+	 * @return mixed|null
+	 */
+	protected function getOption($key) {
+		return (isset($this->options[$key])) ? $this->options[$key] : null;
+	}
```

Comment:
Yes, they are implementation detail. A signal should offer a public method for setting any options and then internally store them in the `$options` array. This also allows type hinting / checking the passed parameter. Example: https://github.com/studer-raimann/ILIAS/blob/9a8447540af74d8d6057e19286b0b255c9d30c70/src/UI/Implementation/Component/Popover/ReplaceContentSignal.php#L20